src/components/Steps/Steps.tsx

```diff
@@ -33,13 +37,35 @@ const Steps = ({
     'd-sm-none': collapse !== true,
     'text-center': true,
   });
-
+  const barColor = granular ? '#cdd5db' : '#818a91';
   return (
     <div className="mb-3">
       <ol className={className}>
         {steps.map((name, index) => {
           const stepComplete = !complete && index < step;
           const stepActive = !complete && index === step;
+          const stepAfterActive = !complete && index === step + 1;
+          const progressPercent = 50;
+          const progressColor = '#016eb7';
+          const renderActiveProgress =
+            granular && !vertical && stepActive && index < steps.length - 1;
+          const renderPrevStepProgress =
+            granular && !vertical && stepAfterActive && index < steps.length && stepProgress > 50;
+          const progressStyle = {
+            backgroundImage: renderActiveProgress
+              ? `linear-gradient(to right, ${
+                  step === 0 ? 'transparent' : progressColor
+                } 50%, ${progressColor} 50%, ${progressColor} ${
+                  progressPercent + stepProgress
+                }%, ${barColor} ${progressPercent + stepProgress}%)`
+              : renderPrevStepProgress
+              ? `linear-gradient(to right, ${progressColor} ${
+                  stepProgress - progressPercent
+                }%, ${barColor} ${stepProgress - progressPercent}%, ${barColor} 50%, ${
+                  index === steps.length - 1 ? 'transparent' : barColor
+                } 50%)`
+              : '',
+          };
```

Comment:
I think doing this in javascript was not necessary. It's a bit complicated to read through this and I think it'd be a lot simpler if you let CSS handle this. We can most likely do this in CSS vs javascript.